### PR TITLE
Use new nettest simpleserver from Shipyard

### PR DESCRIPTION
**Pull request by dfarrell07**
The nginx image is giving us maintenance issues, use a simple netcat
process run from an existing container instead.

### CODE REVIEW

**Comment by dfarrell07** [comment]
The upgrade tests are still correctly failing, see https://github.com/submariner-io/lighthouse/pull/736#issuecomment-1111213026 for details.

**Comment by dfarrell07** [comment]
There are _lots_ of nginx references in Lighthouse. To properly move away from it, even in names, would be a major change.

```
[~/go/src/submariner-io/lighthouse]$ grep -riI --exclude-dir=vendor nginx . | wc -l
324
```

Also wondering if one of these usages is something the `nc` nifty hack doesn't support?

**Comment by dfarrell07** [comment]
The E2E failures here are very similar to the ones on Mike's PR with only the Shipyard version bump.

> Failed to  service IP verification. expected execution result "10.2.96.4" to be empty

https://github.com/submariner-io/lighthouse/runs/6198528492?check_suite_focus=true#step:3:2239

> Failed to retrieve Endpoints. endpoints have 0 IPs when expected 3

https://github.com/submariner-io/lighthouse/runs/6198528492?check_suite_focus=true#step:3:2295

> Failed to  service IP verification. expected execution result "10.2.96.4\n10.1.0.6" to be empty

https://github.com/submariner-io/lighthouse/runs/6198528492?check_suite_focus=true#step:3:2584

**Review by mkolesnik — DISMISSED** [review]
Can you rebase on top of #738 instead of #735 ?

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.12.1
 	github.com/submariner-io/admiral v0.12.0-m3.0.20220223161649-65232cddf5c0
-	github.com/submariner-io/shipyard v0.12.0-m3.0.20220421170648-adf7571e55ed
+	github.com/submariner-io/shipyard v0.12.0-m3.0.20220427131749-6b99f918e45e
```
Comment (mkolesnik) [inline go.mod]: No need to do this commit, the image is available through Quay and you're just using it

File: test/e2e/framework/framework.go [hunk 2]
```diff
@@ -394,15 +394,15 @@ func (f *Framework) NewNginxStatefulSet(cluster framework.ClusterIndex) *appsv1.
 					Containers: []v1.Container{
 						{
 							Name:            statefulServiceName,
-							Image:           "quay.io/testing-farm/nginx:latest",
+							Image:           "localhost:5000/nettest:local",
```
Comment (mkolesnik) [inline test/e2e/framework/framework.go]: This assumes that it was loaded to a local registry which might not be true when calling `subctl verify`, better use `quay.io/submariner/nettest:devel`

**Review by aswinsuryan — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Comment by dfarrell07** [comment]
This is backported here: https://github.com/submariner-io/lighthouse/pull/748

**Pull request merged**